### ISSUE TITLE
[Bug] Mac: "Merge All Windows" fails if focused window is borderless

### ISSUE THREAD

**Issue by dannye** [fltk/fltk#1100]
**Describe the bug**

If the program contains at least 2 windows and the focused window is borderless, clicking "Merge All Windows" in the sys menu hides the borderless window without combining the two windows.

**To Reproduce**

```diff
diff --git a/test/fullscreen.cxx b/test/fullscreen.cxx
index 65c25036a..b5c5009ad 100644
--- a/test/fullscreen.cxx
+++ b/test/fullscreen.cxx
@@ -372,5 +372,10 @@ int main(int argc, char **argv) {
   window.end();
   window.show(argc,argv);
 
+  Fl_Single_Window second_window(100, 100, 200, 200, "Second Window");
+  second_window.size_range(200, 200);
+  second_window.end();
+  second_window.show();
+
   return Fl::run();
 }
```

1. Apply the above patch to test/fullscreen.cxx
2. Build like normal and run test/fullscreen
3. Click the "Border" button to remove the border from the first window
4. Select "Merge All Windows" from the Window menu (make sure the borderless window still has focus first)
5. See that the borderless window is gone and that the second window is unchanged

**Expected behavior**

"Merge All Windows" should probably do nothing in this case (just like what happens if "Merge All Windows" is selected while the second window has focus).

**Video**

https://github.com/user-attachments/assets/d3eb6e33-e475-42c9-bc45-9a71fd3303cc

**FLTK Version**
 - Version: 1.4.0
 - Downloaded and self-built from: Git
 - If from Git, branch: master
 - If from Git, commit: 7ce5b74

**FLTK Configure / Build Options**
 - `cmake -D CMAKE_BUILD_TYPE=Debug`

**Operating System / Platform:**
 - OS: macOS
 - OS Version: macOS 14.6.1
 - Processor if relevant: M3

**Comment by MatthiasWM** [comment]
As a daily user, this is the first time I see this menu item, even though it seems to exist since macOS 10.12 . FLTK does not have an API to group windows into a useful tab group, and AFAIK, no other platform supports this. I believe we should just disable this menu item, that is created by macOS by default. Looking at FLUID as an example, it would not make any sense to merge the app window, the source preview, the tool bar, and custom windows into a single windows with tabs.

https://developer.apple.com/documentation/appkit/nswindow/1646657-allowsautomaticwindowtabbing

**Comment by dannye** [comment]
@MatthiasWM I believe this was added in FLTK 1.4.

See `Fl_Sys_Menu_Bar::create_window_menu()`:

https://github.com/fltk/fltk/blob/7ce5b74e12103ab5ec372ecc2500435deb22d7aa/src/Fl_Sys_Menu_Bar.cxx#L226-L243

It actually works really robustly. This is one of only a few quirks that I've found with it so far.

**Comment by MatthiasWM** [comment]
Huh. Live and learn. Thanks for the link!

**Comment by dannye** [comment]
"Merge All Windows" is actually very nice for FLUID too, because it allows the "Help" window to be put as a tab beside the main editor:

<img width="899" alt="image" src="https://github.com/user-attachments/assets/f786be9a-a8ba-431d-8fc9-e5ddf9e1ff39">

(This is specifically how I've always tested "Merge All Windows" with my editor as well.)

**Comment by ManoloFLTK** [comment]
@MatthiasWM  Window tabbing-related items of the Window menu were introduced in FLTK's macOS platform (by me) to try to make FLTK apps behave as good macOS citizens. Today's document-related Apple apps (_e.g._, TextEdit, Preview) contain the "Merge All Windows" item.

I agree that this item, which is part of the standard macOS HI, is not very convenient when an app handles diverse kinds of windows. In my app, I have defined diverse window classes and assign one to each window with `Fl_Window::xclass()`. I replaced "Merge All Windows" by an item called "Merge Windows by Class" which merges only windows of the same class. Such an operation requires one Objective-c instruction to send the `addTabbedWindow:ordered:` message to a window. We could (in 1.5) add support for this to FLTK.
